ste/sourceInfoProvider-Local_windows.go

```diff
@@ -62,20 +61,32 @@ func (f localFileSourceInfoProvider) GetSDDL() (string, error) {
 	}
 	buf := make([]byte, 512)
 	bufLen := uint32(len(buf))
+	needValidate := false
 	status := ntdll.CallWithExpandingBuffer(func() ntdll.NtStatus {
-		return ntdll.NtQuerySecurityObject(
+		status := ntdll.NtQuerySecurityObject(
 			fd,
 			windows.OWNER_SECURITY_INFORMATION|windows.GROUP_SECURITY_INFORMATION|windows.DACL_SECURITY_INFORMATION,
 			(*ntdll.SecurityDescriptor)(unsafe.Pointer(&buf[0])),
 			uint32(len(buf)),
 			&bufLen)
+
+		// mitigate weirdness with 0 buflen on success with a NAS source
+		if status == ntdll.STATUS_SUCCESS && bufLen == 0 {
+			bufLen = uint32(len(buf))
+			needValidate = true
+		}
+
+		return status
 	}, &buf, &bufLen)
 
 	if status != ntdll.STATUS_SUCCESS {
-		return "", errors.New(fmt.Sprint("failed to query security object", f.jptm.Info().Source, "ntstatus:", status))
+		return "", fmt.Errorf("failed to query security object %s (ntstatus: %s)", f.jptm.Info().Source, status.String())
 	}
 
 	sd := (*windows.SECURITY_DESCRIPTOR)(unsafe.Pointer(&buf[0])) // ntdll.SecurityDescriptor is equivalent
+	if needValidate && !sd.IsValid() {
```

Comment:
This could have been done unconditionally.

Reply:
Technically we probably don't even need to do it.

I discussed with Jason Shay, and from like Vista back, 0 tends to be what gets returned, even if a valid value gets returned.

That said, I felt like best practice was to check it if things didn't seem completely right.